File: src/main/java/com/example/backend/model/Users/UserResponse.java
package com.example.backend.model.Users;

import com.example.backend.model.roles.RoleRequest;
import com.example.backend.model.roles.UserRoleResponse;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.util.List;
import java.util.stream.Collectors;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserResponse {
    private Long id;
    private String email;
    private String username;
    private String firstName;
    private String lastName;
    private String publicKey;
    private List<UserRoleResponse> roles;



    public static UserResponse returnUser(Users users) {
        return new UserResponse(
                users.getId(),
                users.getEmail(),
                users.getUsername(),
                users.getFirstName(),
                users.getLastName(),
                users.getPublicKey(),
                users.getUserRoles().stream().map(UserRoleResponse::fromUserRole).collect(Collectors.toList())
        );
    }

}

File: app/src/main/java/com/example/frontend_android/model/Users/UserResponse.kt
package com.example.frontend_android.model.Users

import com.example.frontend_android.model.roles.RoleResponse


data class UserResponse(
     val id: Long,
     val email: String,
     val username: String,
     val firstName: String,
     val lastName: String,
     val publicKey: String?,
     val roles: List<RoleResponse>
)
